### PR TITLE
Develop chapter5

### PR DESCRIPTION
**Pull request by SorinMsgZ**
CSV Translator

### CODE REVIEW

File: src/test/java/ro/msg/learning/shop/controller/integrationTest/StockExportControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+        productCategoryService.create(productCategoryOne);
+        productCategoryService.create(productCategoryTwo);
+
+        SupplierDTO supplierOne = new SupplierDTO("Fisher-Huels");
+        SupplierDTO supplierTwo = new SupplierDTO("D Amore, Torp and Kuvalis");
+        supplierService.create(supplierOne);
+        supplierService.create(supplierTwo);
+
+        ProductDTO productOne = new ProductDTO();
+        productOne.setId(1);
```
Comment (oana-andreea) [inline src/test/java/ro/msg/learning/shop/controller/integrationTest/StockExportControllerTest.java:59]: please use the builder

File: src/test/java/ro/msg/learning/shop/controller/integrationTest/StockExportRestControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+        productCategoryController.create(productCategoryOne);
+        productCategoryController.create(productCategoryTwo);
+
+        SupplierDTO supplierOne = new SupplierDTO("Fisher-Huels");
+        SupplierDTO supplierTwo = new SupplierDTO("D Amore, Torp and Kuvalis");
+        supplierController.create(supplierOne);
+        supplierController.create(supplierTwo);
+
+        ProductDTO productOne = new ProductDTO();
+        productOne.setId(1);
```
Comment (oana-andreea) [inline src/test/java/ro/msg/learning/shop/controller/integrationTest/StockExportRestControllerTest.java:75]: please use the builder

**Review by oana-andreea — CHANGES_REQUESTED** [review]

File: src/main/java/ro/msg/learning/shop/services/StockExportService.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+@Service
+@RequiredArgsConstructor
+public class StockExportService {
+
+    private final StockRepository stockRepository;
+
+    public List<StockExportDTO> exportingStockByLocationId(int locationId) {
+        return stockRepository.findAll().stream()
+                .map(StockExportDTO::new).filter(stockDTO -> stockDTO.getLocationId() == locationId)
```
[the remark this answers is not in the document]
Comment (SorinMsgZ) [reply src/main/java/ro/msg/learning/shop/services/StockExportService.java]: Done

File: src/test/java/ro/msg/learning/shop/controller/integrationTest/StockExportControllerTest.java (hunk 1 above)
Comment (SorinMsgZ) [reply src/test/java/ro/msg/learning/shop/controller/integrationTest/StockExportControllerTest.java]: Done

File: src/test/java/ro/msg/learning/shop/controller/integrationTest/StockExportRestControllerTest.java (hunk 2 above)
Comment (SorinMsgZ) [reply src/test/java/ro/msg/learning/shop/controller/integrationTest/StockExportRestControllerTest.java]: Done

**Comment by SorinMsgZ** [comment]
PR2A - CSV translator Stock Export - FINISH

**Pull request merged**